### PR TITLE
Fix covariance matrix expansion for duplicate parameters

### PR DESCRIPTION
**Pull request by QRemy**
Fix covariance matrix expansion for duplicate parameters
Resolve issue #2859

### CODE REVIEW

**Comment by QRemy** [comment]
The datasets IO test fails because it do not read the new version of the models.yaml file updated in gammapy-extra.

File: gammapy/datasets/tests/test_io.py [hunk 1]
```diff
@@ -59,3 +64,9 @@ def test_datasets_to_io(tmp_path):
     assert dataset0.psf is not None
     assert dataset0.edisp is not None
     assert_allclose(dataset0.background_model.evaluate().data.sum(), 4094.2, atol=0.1)
+
+    Fit(datasets).run()
```
Comment (registerrier) [inline gammapy/datasets/tests/test_io.py:68]: I think this now requires a `@requires_minuit`.

**Comment by registerrier** [comment]
I have updated the gammapy-data repo with the correct yaml file. Tests are now running.

File: gammapy/datasets/tests/test_io.py (hunk 1 above)
Comment (QRemy) [reply gammapy/datasets/tests/test_io.py:69]: added @requires_dependency("iminuit")

**Pull request merged**